rust/ethers-prometheus/src/json_rpc_client.rs

```diff
@@ -172,3 +185,32 @@ where
         res
     }
 }
+
+impl<C: JsonRpcClient + 'static> From<PrometheusJsonRpcClient<C>> for Box<dyn BlockNumberGetter> {
+    fn from(val: PrometheusJsonRpcClient<C>) -> Self {
+        Box::new(JsonRpcBlockGetter::new(val))
+    }
+}
+
+/// Utility struct for implementing `BlockNumberGetter`
+#[derive(Debug, new)]
+pub struct JsonRpcBlockGetter<T: JsonRpcClient>(T);
```

Comment:
I feel like this & the content below it doesn't feel right to put in `ethers-prometheus` as it's not directly related to ethers-prometheus.

Maybe somewhere in `chains/hyperlane-ethereum/src/rpc_clients`? Or was this motivated by avoiding circular imports? Just feels weird to have this here, but I see that `chains/hyperlane-ethereum/src/rpc_clients/fallback.rs` already assumes that the PrometheusJsonRpcClient is being used, so if it's a huge hassle feel free to leave this as-is

Reply:
The only way I can move it out of here is if we wrapped `PrometheusJsonRpcClient` in a newtype in `hyperlane-ethereum` (and maybe pulled the Deref-on-newtype cheat to minimize changes). That's due to the orphan rule, since this is where `PrometheusJsonRpcClient` is defined and this trait impl will fail: `impl From<PrometheusJsonRpcClient<C>> for Box<dyn BlockNumberGetter>`